test/e2e/resource_fairness/resource_fairness_test.go

```diff
@@ -191,14 +248,203 @@ var _ = Describe("FairScheduling:", func() {
+			RequiredNode: workerID2,
+			Labels:       map[string]string{constants.LabelQueueName: queuePath}}
+
+		// Verify if pod3 and pod4 are assigned based on node sorting fairness.
+		sleepPod3Conf := k8s.SleepPodConfig{
+			Name: "pod3", NS: ns, AppID: "app3",
+			Labels: map[string]string{constants.LabelQueueName: queuePath}}
+		sleepPod4Conf := k8s.SleepPodConfig{
+			Name: "pod4", NS: ns, AppID: "app4",
+			Labels: map[string]string{constants.LabelQueueName: queuePath}}
+
+		sleepPodConfs := []k8s.SleepPodConfig{}
+		sleepPodConfs = append(sleepPodConfs, sleepPod1Conf)
+		sleepPodConfs = append(sleepPodConfs, sleepPod2Conf)
+		sleepPodConfs = append(sleepPodConfs, sleepPod3Conf)
+		sleepPodConfs = append(sleepPodConfs, sleepPod4Conf)
+
+		// The node where the pod is expected to be deployed affects its resource utilization.
+		expectedNode := []string{
+			workerID1,
+			workerID2,
+			workerID1,
+			workerID2,
+		}
+		increaseUtiliz := []map[string]float64{
+			{"vcore": 0.05, "memory": 0.10},
+			{"vcore": 0.10, "memory": 0.05},
+			{"vcore": 0.15, "memory": 0.15},
+			{"vcore": 0.15, "memory": 0.15},
+		}
+		// Deploy pod
+		for idx, config := range sleepPodConfs {
+			ginkgo.By("Deploy the sleeppod " + config.Name)
+			config.CPU = fillNodeUtil(expectedNode[idx], "vcore", increaseUtiliz[idx]["vcore"]+basicUtiliz)
+			config.Mem = fillNodeUtil(expectedNode[idx], "memory", increaseUtiliz[idx]["memory"]+basicUtiliz) / 1000 / 1000
+			initPod, podErr := k8s.InitSleepPod(config)
+			Ω(podErr).NotTo(HaveOccurred())
+			_, err = kClient.CreatePod(initPod, ns)
+			Ω(err).NotTo(HaveOccurred())
+			err = kClient.WaitForPodRunning(ns, config.Name, 30*time.Second)
+			Ω(err).NotTo(HaveOccurred())
+		}
+
+		// Verify pod3, pod4 have been deployed on the correct node.
+		ginkgo.By(sleepPod3Conf.Name + " should deploy on " + workerID1)
+		RespPod, err := kClient.GetPod(sleepPod3Conf.Name, ns)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(RespPod.Spec.NodeName).To(gomega.Equal(workerID1), "Pod should place on "+workerID1)
+
+		ginkgo.By(sleepPod4Conf.Name + " should deploy on " + workerID2)
+		RespPod, err = kClient.GetPod(sleepPod4Conf.Name, ns)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(RespPod.Spec.NodeName).To(gomega.Equal(workerID2), "Pod should place on "+workerID2)
+	})
+
 	AfterEach(func() {
 		testDescription := ginkgo.CurrentSpecReport()
 		if testDescription.Failed() {
 			tests.LogTestClusterInfoWrapper(testDescription.FailureMessage(), []string{ns})
 			tests.LogYunikornContainer(testDescription.FailureMessage())
 		}
-		By("Tear down namespace: " + ns)
-		err := kClient.TearDownNamespace(ns)
-		Ω(err).NotTo(HaveOccurred())
+
+		// Delete all sleep pods
+		ginkgo.By("Delete all sleep pods")
+		err := kClient.DeletePods(ns)
+		if err != nil {
+			fmt.Fprintf(ginkgo.GinkgoWriter, "Failed to delete pods in namespace %s - reason is %s\n", ns, err.Error())
+		}
 	})
 })
+
+func fillNodeUtil(nodeId string, resourceType string, percent float64) int64 {
```

Comment:
Rename the function, it calulates something, but what? I suppose it's something based on a percentage. It could be sth like `getFillResourceValue()`.
Add a comment to make this clear.